ImagePicker/ImagePickerDataSource.swift

```diff
@@ -57,31 +47,32 @@ final class ImagePickerDataSource : NSObject, UICollectionViewDataSource {
             return cell
             
         case 2:
+            return assetItems(for: indexPath, collectionView: collectionView, cellsRegistrator: cellsRegistrator)
             
-            let asset = assetsModel.fetchResult.object(at: indexPath.item)
-            let cellId = cellsRegistrator.cellIdentifier(forAsset: asset.mediaType) ?? cellsRegistrator.cellIdentifierForAssetItems
-            
-            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellId, for: indexPath) as? ImagePickerAssetCell else {
-                fatalError("asset item cell must conform to \(ImagePickerAssetCell.self) protocol")
-            }
-            
-            let thumbnailSize = assetsModel.thumbnailSize ?? .zero
-            
-            // Request an image for the asset from the PHCachingImageManager.
-            cell.representedAssetIdentifier = asset.localIdentifier
-            assetsModel.imageManager.requestImage(for: asset, targetSize: thumbnailSize, contentMode: .aspectFill, options: nil, resultHandler: { image, _ in
-                // The cell may have been recycled by the time this handler gets called;
-                // set the cell's thumbnail image only if it's still showing the same asset.
-                if cell.representedAssetIdentifier == asset.localIdentifier && image != nil {
-                    cell.imageView.image = image
-                }
-            })
-            
-            return cell as! UICollectionViewCell
-        
         default: fatalError("only 3 sections are supported")
         }
     }
     
-
+    private func assetItems(for indexPath: IndexPath, collectionView: UICollectionView, cellsRegistrator: CellRegistrator) -> UICollectionViewCell {
+        let asset = assetsModel.fetchResult.object(at: indexPath.item)
+        let cellId = cellsRegistrator.cellIdentifier(forAsset: asset.mediaType) ?? cellsRegistrator.cellIdentifierForAssetItems
+        
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellId, for: indexPath) as? ImagePickerAssetCell else {
+            fatalError("asset item cell must conform to \(ImagePickerAssetCell.self) protocol")
```

Comment:
assertion might be better - as dependency we don't want to crash the app, we want to notify developer he did something wrong

Reply:
wont do